clcache.py

```diff
@@ -74,6 +80,80 @@ def __str__(self):
         return repr(self.message)
 
 
+class ManifestsManager(object):
+    HASH_INVALIDATION_COUNTER = 3
+
+    def __init__(self, manifestsRootDir):
+        self._manifestsRootDir = manifestsRootDir
+
+    def manifestDir(self, manifestHash):
+        return os.path.join(self._manifestsRootDir, manifestHash[:2])
+
+    def manifestPath(self, manifestHash):
+        return os.path.join(self.manifestDir(manifestHash), manifestHash + ".json")
+
+    def setManifest(self, manifestHash, manifest):
+        ensureDirectoryExists(self.manifestDir(manifestHash))
+        with open(self.manifestPath(manifestHash), 'w') as outFile:
+            # Convert namedtupel to dict to get key names
+            doc = {
+                'includeFiles': manifest.includeFiles,
+                'includesContentToObjectMap': manifest.includesContentToObjectMap
+            }
+            json.dump(doc, outFile, sort_keys=True, indent=2)
+
+    def getManifest(self, manifestHash):
+        fileName = self.manifestPath(manifestHash)
+        if not os.path.exists(fileName):
+            return None
+        try:
+            with open(fileName, 'r') as inFile:
+                doc = json.load(inFile)
+                return Manifest(doc['includeFiles'], doc['includesContentToObjectMap'])
+        except IOError:
+            return None
+
+    def clean(self, maxManifestsSize):
+        manifestFiles = []
+        for path, _, filenames in WALK(self._manifestsRootDir):
+            for filename in filenames:
+                manifestFiles.append(os.path.join(path, filename))
```

Comment:
If the same code is used for cleaning, too, then maybe this would make a nice general purpose generator like:

```python
def filesBeneath(path):
    for path, _, filenames in WALK(path):
        for filename in filenames:
            yield os.path.join(path, filename)
```

...because then you can write

```python
manifestFiles = filesBeneath(self._manifestsRootDir)
```

which looks kinda nice.